spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractReplyProducingMessageHandler.java

```diff
@@ -91,6 +92,11 @@ protected ClassLoader getBeanClassLoader() {
 		return this.beanClassLoader;
 	}
 
+	@Override
+	public IntegrationPatternType getIntegrationPatternType() {
+		return IntegrationPatternType.outbound_gateway;
```

Comment:
This is not intuitive; I would expect the default to be service_activator; I am guessing you did this to avoid having to patch every outbound gateway, which is ok, but perhaps a comment here to explain why this is the default might be useful.

Reply:
Right. I did an opposite: whenever it is not an `outbound gateway`, I did an explicit type: `service activator`, `transformer`, `splitter` etc.

Reply:
Right, but I think it warrants a comment (not java doc, just `/* ... */`.